lib/common/utils.ts

```diff
@@ -128,24 +128,25 @@ export function patchProperty(obj: any, prop: string) {
   };
 
   Object.defineProperty(obj, prop, desc);
-};
+}
 
 export function patchOnProperties(obj: any, properties: string[]) {
-  const onProperties = [];
-  for (const prop in obj) {
-    if (prop.substr(0, 2) == 'on') {
-      onProperties.push(prop);
-    }
-  }
-  for (let j = 0; j < onProperties.length; j++) {
-    patchProperty(obj, onProperties[j]);
-  }
   if (properties) {
     for (let i = 0; i < properties.length; i++) {
       patchProperty(obj, 'on' + properties[i]);
     }
+  } else {
```

Comment:
@vikerman , I think we should merge the `properties` array and `prop in obj`  and then remove duplicate ones, because if we use `if/else` here, some properties may not be patched.

so I think the logic should be.
```
 if (properties) {
   for (let i = 0; i < properties.length; i++) {
      const onProp = 'on' + properties[i];
      if (onProperties.filter(op => op === onProp)).length > 0) {
          continue;
      } 
      patchProperty(obj, onProp);
    }
  }
```